### PR TITLE
Kostas/ops 11290

### PR DESCRIPTION
**Pull request by Kntanos**

### CODE REVIEW

File: app/views/calculator/select_a_payment_plan_form.scala.html [hunk 1]
```diff
@@ -121,13 +115,24 @@
...
+@viewsHelpers.form(redirectCall, Symbol("autocomplete") -> "off") {
     <fieldset class="govuk-fieldset">
-        <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
-            <h1 class="govuk-heading-xl">@Messages("ssttp.calculator.results.title")</h1>
-        </legend>
-
-        <p class="govuk-body">@Html(messages(s"ssttp.calculator.results.p1"))</p>
+            <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+                <h1 class="govuk-heading-xl">
+                    @{previousFormSubmission.fold(messages("ssttp.calculator.results.title"))(_.selection match {
```
Comment (AndyHWChung) [inline app/views/calculator/select_a_payment_plan_form.scala.html]: you could probably avoid doing the same fold and pattern match twice by somehow having the correct legend and p under the right cases in the pattern match. The twirl syntax to get that working might take a couple of goes to get right

Comment (Kntanos) [reply app/views/calculator/select_a_payment_plan_form.scala.html]: I tried it. It generates so much more code and you still have to repeat all the html elements three times, one Right and twice for Left. I think it is harder to read too. But I'm happy if you prefer it. It looks like this:

```scala
 @{previousFormSubmission.fold(
                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                    <h1 class="govuk-heading-xl">
                    {messages("ssttp.calculator.results.title")}
                    </h1>
                </legend>
                <p>{Html(messages(s"ssttp.calculator.results.p1"))}</p>
        )(_.selection match {
            case Right(CustomPlanRequest(_)) =>
                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                    <h1 class="govuk-heading-xl">
                        {messages("ssttp.calculator.results.title.custom-amount")}
                    </h1>
                </legend>
                <p>{Html(messages(s"ssttp.calculator.results.p1.custom-amount"))}</p>
            case Left(SelectedPlan(amount)) => if(amount > minCustomAmount)
                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                    <h1 class="govuk-heading-xl">
                        {messages("ssttp.calculator.results.title.custom-amount")}
                    </h1>
                </legend>
                <p>{Html(messages(s"ssttp.calculator.results.p1.custom-amount"))}</p>
            else
                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                    <h1 class="govuk-heading-xl">
                        {messages("ssttp.calculator.results.title")}
                    </h1>
                </legend>
                <p>{Html(messages(s"ssttp.calculator.results.p1"))}</p>
            })
        }

Comment (AndyHWChung) [reply app/views/calculator/select_a_payment_plan_form.scala.html]: Does it work if you do something like this at the top of the template?
```
@legendAndP1Text = @{
  previousFormSubmission.fold(
    messages("ssttp.calculator.results.title") -> messages(s"ssttp.calculator.results.p1")
  )(_.section match { ... }
}
```
Then you could extract out the legend text and the p1 text using `legendAndP1Text._1` and `legendAndP1Text._2` and have the html tags only once still

Comment (Kntanos) [reply app/views/calculator/select_a_payment_plan_form.scala.html]: Yes that's clever

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**